src/transformers/__init__.py

```diff
@@ -1775,6 +1779,14 @@
             "NystromformerPreTrainedModel",
         ]
     )
+    _import_structure["models.oneformer"].extend(
+        [
+            "ONEFORMER_PRETRAINED_MODEL_ARCHIVE_LIST",
+            "OneFormerForUniversalSegmentation",
```

Comment:
Regarding the name of the class - it seems that, simiilar to MaskFormer and Mask2Former, OneFormer outputs a set of classes with corresponding binary masks, i.e. it uses the same paradigm as MaskFormer; namely seeing instance/semantic/panoptic segmentation as an instance segmentation problem (i.e. predicting a set of classes + corresponding masks).

If also the postprocessing is identical to MaskFormer, I'd opt to call the class `OneFormerForInstanceSegmentation`.

Reply:
Hi @NielsRogge, it's true that OneFormer also outputs a set of classes + corresponding binary masks. However, if we name the model `OneFormerforInstanceSegmentation`, it overshadows that the same model (or checkpoint) can be used for all three tasks, which is a big selling point for OneFormer.

Reply:
@praeclarumjj3 I think using `OneFormerForUniversalSegmentation` makes sense over here as other /non-universal segmentation models often take quite different approaches for instance and panoptic segmentation. 

We could still have the `post_process_semantic_segmentation`, `post_process_instance_segmentation`, `post_process_panoptic_segmentation` methods within the `ImageProcessor` class. What do you think @NielsRogge ?

Reply:
Hi @alaradirik, thanks for your comment. I did add the `post_process_{task}` methods inside the `OneFormerImageProcessor` class [here](https://github.com/praeclarumjj3/transformers/blob/cb9cba1bf6d0249401ffacfbe9eca54ba1c384c8/src/transformers/models/oneformer/image_processing_oneformer.py#L1086-L1343). Do you want me to do something else?